.github/workflows/build.yml

```diff
@@ -1,4 +1,4 @@
-name: build eosio contracts
+name: Build
```

Comment:
if I got it right the webapp build will run even if there are no relevant changes to it (same for box), is it desired? or did I miss something? (if I'm correct are about to introduce the huge time builds as we had in my previous company which makes me super sad 😢 )

Reply:
![image](https://user-images.githubusercontent.com/418464/128512018-245f5ada-3547-471d-9157-0f347f69b6c5.png)

The problem is this: box now includes `eden-micro-chain.wasm` and GH doesn't currently provide a way for 1 workflow to depend on another.

I see an additional way to speed it up though. `box-build` doesn't use `box-e2e`'s outputs. Likewise `webapp-build` doesn't use `webapp-e2e`'s outputs. I could switch all 4 of those to only depend on `build-micro-chain` to increase parallelism.

Reply:
Does anyone use `webapp-build`'s output?

Reply:
Without conditionals:

![image](https://user-images.githubusercontent.com/418464/128534024-452dbbc2-a5e4-4cb4-af1e-42dfd7f74cec.png)

Reply:
The JS ones all run in parallel

Reply:
Conditionals are now in place